src/kaleidoscope/key_defs.h

```diff
@@ -77,6 +79,29 @@ class Key {
            );
   }
 
+  constexpr bool isKeyboardKey() const {
```

Comment:
the "isKeyboard" prefix feels a little funny to me. 

isModifier makes tons of sense to me.

isKeyboardShift feels like a kind of strange thing to be breaking out. Can you tell me a bit more about it?

Reply:
Sure. The `isKeyboard*` prefix is meant to convey the HID type of the key (as opposed to Consumer & System keys, and maybe Mouse keys). It also helps to distinguish from other `Key` types—in particular Layer keys.

`isModifier()` is, to me, not specific enough, because there are other things that "modify" key behaviour, some of which get treated the same way as HID Keyboard modifiers by some plugins. Specifically, layer shift keys, which fall into the same category for at least a few plugins, notably Qukeys and OneShot. `isKeyboardModifier()` implies (and includes) `isKeyboardKey()`, and is, I think, less ambiguous.

Regarding `isKeyboardShift()` (as opposed to `isLayerShift()`, thus the name I chose) — Shift is conceptually special, even among modifiers, because it's the one that changes the visible output character. As such, there are plugins (TopsyTurvy, ShapeShifter are examples) that have a particular interest in whether or not the user is holding an intentional `shift` key. I don't know of any plugins that care specifically about any other modifiers, because those don't get used to affect the keymap (for instance, wanting to map a key that's got both `,` and `;` or `+` and `&`).

One of the reasons I'm pushing for the whole package of changes I've been working on is because it's very difficult with Kaleidoscope as it exists today to write a plugin that will allow me to rearrange non-alphabetic symbols at will to create the keymap that I want. That plugin also needs the same test as TopsyTurvy and ShapeShifter (which, even combined can't give me that freedom to rearrange symbols).

I would use `isKeyboardShift()` in several places, though probably not quite as much as `isKeyboardModifier()` or `isLayerShift()`, and it would be nice to just have it available, and not need to rewrite it each time, possibly making errors.

Reply:
I started to add `isConsumerKey()` & `isSystemKey()` as well, but the only place that comes to mind that would use them already has a more efficient version than could be constructed this way. Also, `isConsumerKey()` would require furthe rearranging of the code the put the consumer key bitmask in the right place.

Reply:
Got it.

I like the concept and the rationales make sense. The names don't feel right to me yet, but I don't have a better idea right now.

I want to look at all of this when I'm not overtired.

Reply:
In case it helps, here's an example of what `isKeyboardShift()` would look like "in the wild":
https://github.com/keyboardio/Kaleidoscope/blob/c3c8eae89cf3d4c2af7a3b43650db9f702ebc974/src/kaleidoscope/plugin/TopsyTurvy.cpp#L55

Reply:
Also, Qukeys and OneShot would have several places with a test that looks like this:
```c++
if (key.isKeyboardModifier() || key.isLayerShift()) {...}
```